src/main/java/io/supertokens/inmemorydb/Start.java

```diff
@@ -3032,9 +3032,27 @@ public void addClientForApp(AppIdentifier appIdentifier, String clientId)
 
             if (isPrimaryKeyError(serverErrorMessage, config.getOAuthClientTable(),
                     new String[]{"app_id", "client_id"})) {
-                throw new ClientAlreadyExistsForAppException();
+                throw new OAuth2ClientAlreadyExistsForAppException();
             }
             throw new StorageQueryException(e);
         }
     }
+
+    @Override
+    public boolean isClientIdAlreadyExists(String clientId) throws StorageQueryException {
+        try {
+            return OAuthQueries.isClientIdAlreadyExists(this, clientId);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public void removeAppClientAssociation(AppIdentifier appIdentifier, String clientId) throws StorageQueryException {
```

Comment:
this should communicate back to the caller if the client was associated with the app in the first place? For example, with a boolean

Reply:
It could, but currently there is no need for it. I would say to do so, when we want to use that for anything.

Reply:
I think it's important to communicate it here cause of Developer experience. If they are deleting a client and by mistake make a typo, then they will get back the same response in case when they have actually deleted a client. So they may be under the wrong impression that the client is deleted (in case they made a typo)